channeld/channeld.c

```diff
@@ -1977,6 +1977,73 @@ static void send_onionmsg(struct peer *peer, const u8 *msg)
 						    onion_routing_packet,
 						    tlvs)));
 }
+
+static void handle_tx_sigs(struct peer *peer, const u8 *msg)
+{
+	struct channel_id cid;
+	struct bitcoin_txid txid;
+	struct witness_stack **ws;
+	const struct wally_tx *wtx;
+	size_t j = 0;
+
+	if (!fromwire_tx_signatures(tmpctx, msg, &cid, &txid, &ws))
+		peer_failed(peer->pps,
+			    &peer->channel_id,
+			    "Bad tx_signatures %s", tal_hex(msg, msg));
+
+	/* Check that we're in the right spot for this channel to have gotten
+	 * this message */
+	if (peer->funding_locked[LOCAL] || peer->funding_locked[REMOTE]) {
+		/* FIXME: should we fail here instead? */
```

Comment:
Yes, please do.  Or at very least, log BROKEN so our test harness will pick it up if we hit it!